### DISCUSSION TITLE
[Question] @rnx-kit/dep-check - Understand why a package's version is recommended

### DISCUSSION THREAD

**Discussion by Thanaen** [microsoft/rnx-kit#1552] [Q&A] [answered]
[NOTE] I wanted to ask this question in the "Discussions" section, but I couldn't find the appropriate category 😅 [NOTE]

There is only one thing that prevents me from using this package: it recommends to use an old version of react-navigation (v5 instead of v6). This is also the case for other packages (rn-reanimated or rn-gesture-handler).
I figure if the tool says so, there must be a reason.
Is there any way to know what the reason is? 

The v6 of rn-navigation brings some changes that are used in my current project, and I don't feel that my application is broken, so I'm curious.

Thanks in advance, and thanks again for this tool! :)

**Comment by kelset** [comment, marked as answer]
👋 @Thanaen!

I think it's a great question - and we should make it clearer in the docs themselves. The basic answer is that what is recommended there is what we use at Microsoft internally; we tend to be a bit "defensive" in how and what we upgrade because of the size of our codebases.
At the moment that's the way we can safely say if a version of a package works for a certain version of RN; but we are working in trying to get at least some of the big libraries themselves to declare which version of RN they support via rnx-kit config, so that there will be less of a need of a "default profile" telling you what to use.

But that shouldn't stop you from upgrading to them - dep-check has a feature which is called [custom profiles](https://microsoft.github.io/rnx-kit/docs/guides/dependency-management#customization), and basically it allows you to define a local list of packages and versions that you want to use "on top of" the default profile we provide. So, for your project, what I would suggest is to just create a custom profile, add the packages+versions you are using already in it, then run dep-check again; that way you can also use it across multiple projects/folders in case you have more than one RN project :)

(also, we have some plans to add to dep-check multiple profiles aside our default MSFT so that you won't necessarily need a local custom profile... we're still thinking on how to do it)

**Comment by tido64** [comment]
Out of curiousity, what version of react-native are you using? I think reanimated and gesture-handler should be fairly up-to-date for the later versions.

As for react-navigation, like @kelset said, we trying to ensure that every combination of capabilities work across all Microsoft apps. That said, I think we can probably bump react-navigation to latest major. I just need to check how it is being used internally and what the breaking changes are.

**Comment by afoxman** [comment]
Hi @Thanaen - good callout on the discussions area. It needed some help! I've added a bunch of topics, opening it up into a proper discussion area 🙏💬

**Comment by Thanaen** [comment]
Thank you for your answers!

@kelset Indeed, custom profiles seem to be a good solution for these situations!

@tido64 Currently the application in question is running react-native 0.67.4, and I was planning to use your tool to facilitate the upgrade to 0.68.2 (since it is mentioned in the RN Upgrade Helper, I wanted to test it!) 

Anyway thanks for your involvement, it motivates me even more to integrate RNX Kit in my projects 😁

**Reply by tido64** [reply]
FYI: It looks like we should be fine with the react-navigation bump. Please check out [1.12.15](https://github.com/microsoft/rnx-kit/releases/tag/%40rnx-kit/dep-check%401.12.15).

**Reply by Thanaen** [reply]
Great, you solved my problem before I even tried to make a custom profile! Thanks a lot 😄
